# episcanpy-paper

**EpiScanpy** (Epigenomics single cell analysis in python) is a toolkit to analyse single-cell open chromatin (scATAC-seq) and single-cell DNA methylation (for example scBS-seq) data. **EpiScanpy** is the epigenomic extension of the very popular scRNA-seq analysis tool **Scanpy** (`Genome Biology <https://doi.org/10.1186/s13059-017-1382-0>`__, 2018) [Wolf18]. For more information, read scanpy `documentation <https://scanpy.readthedocs.io>`__. 


The documentation for epiScanpy is available `here <https://episcanpy.readthedocs.io/en/latest/>`__. If epiScanpy is useful to your research, consider citing `epiScanpy <https://www.biorxiv.org/content/10.1101/648097v1>`__.


Here you can find the analysis of bone marrow cell types measured using scATAC-seq (data from Cusanovich et al. 2018 (CITE)), as well as brain cell types measured using scATAC-seq   (from Cusanovich et al. 2018 (CITE)) and using scRNA-seq (from Saunders et al. 2018 (CITE)). 
Also, we provide the scripts used for the benchmarking of epiScanpy to other scATACseq data processing methods using https://github.com/pinellolab/scATAC-benchmarking implementation (Huidong Chen, Caleb Lareau, Tommaso Andreani, Michael E. Vinyard, Sara P. Garcia, Kendell Clement, Miguel A. Andrade-Navarro, Jason D. Buenrostro & Luca Pinello. Assessment of computational methods for the analysis of single-cell ATAC-seq data. Genome Biology 20, 241 (2019).).
You can also find the dockerfile for installation.

Due to the large size of the data, they are not directly available on Github. Send an email to anna.danese@helmholtz-muenchen.de and we will make them available.
